src/identity/contactMapping.ts

```diff
@@ -247,6 +252,46 @@ function* fetchWalletAddresses(e164Number: string) {
   return Array.from(possibleUserAddresses)
 }
 
+function* fetchWalletAddresses(e164Number: string) {
+  const centralPhoneVerificationEnabled = yield select(centralPhoneVerificationEnabledSelector)
+  if (!centralPhoneVerificationEnabled) {
+    return yield call(fetchWalletAddressesDecentralised, e164Number)
+  }
+
+  try {
+    const address = yield select(walletAddressSelector)
+    const signedMessage = yield call(retrieveSignedMessage)
+
+    const response: Response = yield call(fetch, networkConfig.lookupPhoneNumberUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        authorization: `Valora ${address}:${signedMessage}`,
+      },
+      body: JSON.stringify({
+        phoneNumber: e164Number,
+        clientPlatform: Platform.OS,
+        clientVersion: DeviceInfo.getVersion(),
+      }),
+    })
+
+    if (response.ok) {
+      const { addresses } = yield call([response, 'json'])
+      if (addresses.length === 0) {
```

Comment:
I'm wondering if we should call the decentralized flow even if we have a response from the CPV...
A user could be verified in both systems with different addresses and the same phone number. Maybe it's a weird case that probably won't happen until the migration is done.

Reply:
ah yes, good point. i updated to always check the decentralised mapping. discussing with @jeanregisser , it makes sense to plan to maintain backwards compatibility for the foreseeable future :) thanks!